libdnf5/repo/repo.cpp

```diff
@@ -203,6 +206,31 @@ void Repo::read_metadata_cache() {
     p_impl->downloader->load_local();
 }
 
+void Repo::install_appstream() {
+#ifdef WITH_APPSTREAM
+    std::string repo_id = p_impl->config.get_id();
+    const char * as_basenames[] = {"appstream", "appstream-icons", NULL};
```

Comment:
this might not necessarily work when the appstream data is generated by the `appstream-generator` (instead of the `appstream-builder`), but I cannot test what the generator does in Fedora, because it's simply [broken there](https://bugzilla.redhat.com/show_bug.cgi?id=2325398).

Ideally, the `get_metadata_path()` might return a list of the repo files, which contain certain string, like in this case I'd call it `appstream` and any `*appstream*` files would be processed here, instead of knowing beforehand what file "types" can be used by whichever Appstream library and/or its version.

For example, the appstream-generator might create multiple `appstream-icons` files, each containing the icon size in its name, which is very bad approach, because the consumers need to be updated when the icon sizes change, to keep the list of the possible icon sizes in sync with the others.

Reply:
Note that `appdata` and `appdata-icons` are also used (by e.g. SUSE distributions).

Reply:
Do they use different names, thus patch PackageKit? Or how does that work?

Reply:
I see. I just committed a change, which considers any data type which starts with `appstream` or `appdata` prefix as being related to the Appstream and it's tried to be installed.  It allows to avoid hard-coding all the possibilities for the `appstream-icons-64x64.tar.gz`, .... (or what the `appstream-generator` generates). I asked (through a TODO comment), to move that to the configuration, because I do not know how to do it, hence the comment only.